Importing a function from a module:
        >>> BG = __import__("7-base_geometry").BaseGeometry

Checking the module docstring:
         >>> m = __import__("7-base_geometry").__doc__
         >>> len(m) > 1
         True

Checking the class docstring:
         >>> c = __import__("7-base_geometry").BaseGeometry.__doc__
         >>> len(c) > 1
         True

Checking for the method docstring:
	  >>> mod1 = __import__("7-base_geometry").BaseGeometry.area.__doc__
	  >>> len(mod1) > 1
	  True
	  >>> mod2 = __import__("7-base_geometry").BaseGeometry.integer_validator.__doc__
          >>> len(mod2) > 1
          True

Checking for the area failure method:
	  >>> b = BG()
	  >>> b.area()
	  Traceback (most recent call last):
	  ...
	  Exception: area() is not implemented

Checking for too many arguments for area:
	  >>> b.area(1)
	  Traceback (most recent call last):
          ...
	  TypeError: BaseGeometry.area() takes 1 positional argument but 2 were given

Checking the integer validator for passing the integer:
	  >>> b.integer_validator("integer", 1)

Checking for the integer == 0:
	  >>> b.integer_validator("integer", 0)
	  Traceback (most recent call last):
          ...
          ValueError: integer must be greater than 0

Checking for the integer < 0;
	  >>> b.integer_validator("integer", -7)
	  Traceback (most recent call last):
          ...
          ValueError: integer must be greater than 0

Checking for the non-integer types:
	  >>> b.integer_validator("bool", True)
	  Traceback (most recent call last):
          ...
          TypeError: bool must be an integer
	  >>> b.integer_validator("float", 1.7)
          Traceback (most recent call last):
          ...
          TypeError: float must be an integer
	  >>> b.integer_validator("complex", complex(3, 3))
          Traceback (most recent call last):
          ...
          TypeError: complex must be an integer
	  >>> b.integer_validator("string", "hi")
          Traceback (most recent call last):
          ...
          TypeError: string must be an integer
	  >>> b.integer_validator("tuple", (7, 8))
          Traceback (most recent call last):
          ...
          TypeError: tuple must be an integer
	  >>> b.integer_validator("list", [2, 3, 4])
	  Traceback (most recent call last):
          ...
          TypeError: list must be an integer
	  >>> b.integer_validator("dict", {"key": "value"})
          Traceback (most recent call last):
          ...
          TypeError: dict must be an integer
	  >>> b.integer_validator("set", {"hi", "world"})
          Traceback (most recent call last):
          ...
          TypeError: set must be an integer
	  >>> b.integer_validator("frozenset", frozenset(["hi", "world"]))
          Traceback (most recent call last):
          ...
          TypeError: frozenset must be an integer
	  >>> b.integer_validator("bytes", b"bytes")
          Traceback (most recent call last):
          ...
          TypeError: bytes must be an integer
	  >>> b.integer_validator("bytearrays", bytearray(b"bytes"))
          Traceback (most recent call last):
          ...
          TypeError: bytearrays must be an integer

Checking for no arguments to the integer_validator:
	  >>> b.integer_validator()
	  Traceback (most recent call last):
          ...
	  TypeError: BaseGeometry.integer_validator() missing 2 required positional arguments: 'name' and 'value'

Checking for only one argument to integer_validator:
	  >>> b.integer_validator("integer")
	  Traceback (most recent call last):
          ...
	  TypeError: BaseGeometry.integer_validator() missing 1 required positional argument: 'value'

Checking for too many arguments:
	  >>> b.integer_validator("integer", 1, 2)
	  Traceback (most recent call last):
          ...
	  TypeError: BaseGeometry.integer_validator() takes 3 positional arguments but 4 were given